src/codegen/compile.ml

```diff
@@ -4536,6 +4545,8 @@ module Arr = struct
      No difference between mutable and immutable arrays.
   *)
 
+  let max_size env = Int32.shift_left 1l 29 (* inclusive *)
```

Comment:
This is because the array iterator did not and does not support arrays longer than around 500 million elements (1<<29)?
Just curious whether we check the case of longer arrays, since the RTS has a higher limit (around 2 billion elements, 1<<31).

Reply:
Hmm, I think that should actually be 
```suggestion
  let max_size env = Int32.shift_left 1l 28 (* inclusive *)
```
if I want max_size to be 2^29 (silly me)